presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
```

Comment:
I checked this:
```
presto> select grouping(a) from (values 1) t(a) group by t.a;
Query 20170315_094523_00002_57vnq failed: line 1:8:
Arguments in GROUPING() must be in the query's GROUP BY columns. Remove columns '["a"]' from GROUPING() or add them to the GROUP BY.
```

if you'd like to leverage #7354, I offer my guidance and help.